product/src/main/java/com/yas/product/exception/NotFoundException.java

```diff
@@ -1,8 +1,21 @@
 package com.yas.product.exception;
 
-public class NotFoundException extends RuntimeException{
+import com.yas.product.utils.MessagesUtils;
 
-  public NotFoundException(final String message) {
-    super(message);
+public class NotFoundException extends RuntimeException {
+
+  private String message;
+
+  public NotFoundException(String errorCode, Object... var2) {
+    this.message = MessagesUtils.getMessage(errorCode, var2);
+  }
+
+  @Override
```

Comment:
I think we should expose the error code. We might need to use them to compare them when handle exception